### PR TITLE
Navigation: Add useSetNavigationParamsHook for setParams in useEffect hook

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: packages/navigation/src/useSetNavigationParams.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * the component to go into endless re-render.
+ * This custom hook is a workaround for this, ensuring that you watch all values passed to set params, but not react-navigation itself.
+ */
+export default function useSetNavigationParams<T: {}>(
+  setParams: SetParams,
+  config: T,
+) {
+  React.useEffect(() => {
+    setParams(config);
+  }, Object.entries(config).map(([key, value]) => value)); // eslint-disable-line react-hooks/exhaustive-deps, no-unused-vars
```
Comment (RobinCsl) [inline packages/navigation/src/useSetNavigationParams.js]:
```suggestion
  }, Object.values(config)); // eslint-disable-line react-hooks/exhaustive-deps
```

File: app/hotels/src/navigation/allHotels/SearchResultsScreen.js [hunk 2]
```diff
@@ -44,20 +46,15 @@ type Props = {|
 const noop = () => {};
 
 const SearchResultsScreen = (props: Props) => {
-  React.useEffect(() => {
-    props.navigation.setParams({
-      toggleMap: props.setResultType,
-      setCheckinDate: props.setCheckinDate,
-      setCheckoutDate: props.setCheckoutDate,
-    });
-  }, []);
```
Comment (RobinCsl) [inline app/hotels/src/navigation/allHotels/SearchResultsScreen.js]: So with the new implementation, all changes to the values are going to trigger a new `useEffect`, when the previous one set `toggleMap`, `setCheckinDate` and `setCheckoutDate` at the initial render only, and not in subsequent re-renders.
Is that a "regression" in some sense?

Comment (tbergq) [reply app/hotels/src/navigation/allHotels/SearchResultsScreen.js]: After doing some research on the eslint-rule, and specifically this [comment](https://github.com/facebook/react/issues/14920#issuecomment-470863122) I think we should have all props in the array. 

None of this props will change, but in theory they could.

Comment (RobinCsl) [reply app/hotels/src/navigation/allHotels/SearchResultsScreen.js]: Okay, makes sense! Thanks for sharing your research!

**Pull request merged**